File: app/src/main/java/cn/invonate/ygoa3/Util/pingyin/PinyinComparator3.java
package cn.invonate.ygoa3.Util.pingyin;

import java.util.Comparator;

import cn.invonate.ygoa3.Entry.LocalContacts;

/**
 *
 *
 */
public class PinyinComparator3 implements Comparator<LocalContacts> {

    @Override
    public int compare(LocalContacts o1, LocalContacts o2) {
        if (o1.getSortLetters().equals("@")
                || o2.getSortLetters().equals("#")) {
            return -1;
        } else if (o1.getSortLetters().equals("#")
                || o2.getSortLetters().equals("@")) {
            return 1;
        } else {
            return o1.getSortLetters().compareTo(o2.getSortLetters());
        }
    }

}

File: app/src/main/java/cn/invonate/ygoa3/Util/pingyin/PinyinComparator3.kt
package cn.invonate.ygoa3.Util.pingyin

import cn.invonate.ygoa3.Entry.LocalContacts
import java.util.*

/**
 *
 *
 */
class PinyinComparator3 : Comparator<LocalContacts> {

    override fun compare(o1: LocalContacts, o2: LocalContacts): Int {
        return if (o1.sortLetters == "@" || o2.sortLetters == "#") {
            -1
        } else if (o1.sortLetters == "#" || o2.sortLetters == "@") {
            1
        } else {
            o1.sortLetters.compareTo(o2.sortLetters)
        }
    }

}
